### ISSUE TITLE
Dependency Injection with typedi broken, reproducible using examples from type-graphql website

### ISSUE THREAD

**Issue by theseyi** [MichalLytek/type-graphql#840]
**Describe the Bug**
Injecting a service into a Resolver via the constructor, injects the DI Container Instance instead. This happened in my code as well as when trying to reproduce using code from the guide example [using-container](https://github.com/MichalLytek/type-graphql/tree/v1.1.1/examples/using-container)

Instead of getting the Service class injected, the container instance is injected with the classes

`recipeService`
```js
ContainerInstance {services: Array(2), id: "default"}
```

**To Reproduce**
Should be reproducible by running the code examples [using-container](https://github.com/MichalLytek/type-graphql/tree/v1.1.1/examples/using-container) found in the repo
Can create a repo if needed

**Expected Behavior**
Expect the value injected into the constructor to be an instance of  the [RecipeService](https://github.com/MichalLytek/type-graphql/blob/23409fd16fe85ddb2785d06712bdf6f996b89652/examples/using-container/recipe-service.ts)

**Logs**
Inspecting the injected service at runtime: `recipeService`
```js
ContainerInstance {services: Array(2), id: "default"}
```
```json
{
    "services": [
        {
            "global": false,
            "multiple": false,
            "eager": false,
            "transient": false
        },
        {
            "id": "SAMPLE_RECIPES",
            "type": null,
            "global": false,
            "multiple": false,
            "eager": false,
            "transient": false
        }
    ],
    "id": "default"
}
```
```json
{
  "errors": [
    {
      "message": "this.recipeService.getOne is not a function",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "recipe"
      ],
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR",
        "exception": {
          "stacktrace": [
            "TypeError: this.recipeService.getOne is not a function",
            "    at RecipeResolver.recipe (/api/.build/src/handler.js:193434:31)",
            "    at /api/.build/src/handler.js:81240:94",
            "    at processTicksAndRejections (internal/process/task_queues.js:97:5)"
          ]
        }
      }
    }
  ],
  "data": {
    "recipe": null
  }
}
```

**Environment (please complete the following information):**

- OS: macOS 11.2.3
- Node v12.20.1
- Package version 1.1.1
- TypeScript version 4.2.3

**Additional Context**

**Comment by MichalLytek** [comment]
TypeDI examples has been fixed on master branch. I can't edit past commit to fix the code on v1.1.1 tag.

**Comment by theseyi** [comment]
Hey @MichalLytek thanks for looking at this. This is actually happening using the examples on the master branch. Which afaict were fixed by adding the@Service decorator to but the injected class and the injection site class. The injection happens, but the wrong class is injected as mentioned. I'm able to temporarily work around this by doing

constructor(){
  this.service = Container.get('ServiceClass');
}
